var/spack/repos/builtin/packages/paraview/get_version_without_git.patch

```diff
@@ -0,0 +1,29 @@
+--- ParaView-v5.4.1.orig/CMake/ParaViewDetermineVersion.cmake	2017-10-12 14:36:02.000000000 -0600
++++ ParaView-v5.4.1/CMake/ParaViewDetermineVersion.cmake	2017-10-12 14:39:39.000000000 -0600
+@@ -36,16 +36,16 @@
+     # this function.
+     return ()
+   elseif (NOT PARAVIEW_GIT_DESCRIBE)
+-    if(EXISTS ${git_command})
```

Comment:
Again, this looks like it would be more of a hack than me commenting out the specific set of code that uses `git describe` incorrectly, while leaving the rest of the logic intact.

Reply:
I think if we don't want a git version in paraview, we should just tell CMake to not use git, that seem let hackish to me than commenting out a block to code.

And a `git grep GIT_EXECUTALBE` on the Paraview repo shows that `GIT_EXECUTALBE` is only used in `determine_version(${ParaView_SOURCE_DIR} ${GIT_EXECUTABLE} "PARAVIEW")`, so there shouldn't be any side effects. Adding `-DGIT_EXECUTABLE=False` should do it all.

Reply:
Personally I don't feel that grepping for variables in CMake code tells the whole story in that you can programmatically create variables. I do agree it is most likely the case that adding `-DGIT_EXECUTABLE=/not/git` will probably work as intended for this, but this exposes possible side effects I have no idea about, and I am not able to easily test this as Paraview takes hours to compile. We will also be using Catalyst in my project so that makes me adverse to globally changing expected behavior. I agree a patch is not the most flexible way to accomplish this, but I feel it is the *safest* way to accomplish this because it localizes the change to exactly where it is being misused. The file `ParaViewDetermineVersion.cmake` does not change often, so expanding the patch to other versions should be possible, and hopefully they will just fix this in the future because I have submitted this bug to Kitware https://gitlab.kitware.com/paraview/paraview/issues/17761 .

Reply:
We are currently trying to set up   ParaView and Catalyst on our headless cluster ( with Mesa )
We managed to install Catalyst and compile some of the examples.
We did not hit the git problem as our compute nodes  we run the build on have no git installed.
I'm not sure weather just setting -DGIT_EXECUTABLE=/not/git would be completely safe but, according to my experience, Catalyst can be build on a machine without git 
Just my 2 cents

Reply:
Thanks for reporting this!

Reply:
Are we certain that https://github.com/Kitware/ParaView/blob/master/Utilities/Git/Git.cmake won't still find a git executable on the system if I set `-DGIT_EXECUTABLE=false` during CMake configure? If so, I will find time to change this pull request to do so.